crates/red_knot_python_semantic/src/types.rs

```diff
@@ -1220,9 +1279,26 @@ pub struct ClassType<'db> {
     definition: Definition<'db>,
 
     body_scope: ScopeId<'db>,
+
+    is_builtin: Option<BuiltinType>,
 }
 
 impl<'db> ClassType<'db> {
+    /// Find if a class is a builtin type.
+    pub fn maybe_builtin(
```

Comment:
I don't like the name of this method currently because it implies that if it returns `None` the class is not a builtin. But that's not necessarily true, since our list of known builtins isn't exhaustive. I'd prefer `maybe_known_builtin`. Or just `maybe_known` if we go with the broader refactor/rename to handle any known class, even if it's not a builtin.

Reply:
Completely agree. I think I like this `known` prefix - what would you think of also adding it to the type we store internally?
I'm overall afraid that a new reader would come - see a builtin field - and trust that it covers all existing builtins

Reply:
Agreed; I think this is a good enough reason to go ahead and fully make the change to "Known" terminology in this PR, as described in the other comment thread.